src/components/route-history/route-history.js

```diff
@@ -0,0 +1,74 @@
+import {
+  createContext,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+} from 'react';
+import PropTypes from 'prop-types';
+import {
+  useHistory,
+  useLocation,
+} from 'react-router';
+
+const RouteHistoryContext = createContext();
+
+const propTypes = {
+  children: PropTypes.oneOfType([
+    PropTypes.node,
+    PropTypes.arrayOf(PropTypes.node),
+  ]),
+};
+
+const RouteHistoryContextProvider = ({ children }) => {
+  const history = useHistory();
+  const location = useLocation();
+  const routeHistory = useRef(JSON.parse(sessionStorage.getItem('eholdings-history')) || []);
+
+  useEffect(() => {
+    const unlisten = history.listen((_location, action) => {
+      if (action === 'POP') {
+        routeHistory.current.shift();
+      } else {
+        routeHistory.current.unshift(_location);
+      }
+    });
+
+    const onUnmount = () => {
+      unlisten();
+      sessionStorage.setItem('eholdings-history', JSON.stringify(routeHistory.current));
+    };
+
+    return onUnmount;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const navigateBack = useCallback(() => {
+    const nonEholdingsPagesCount = routeHistory.current.findIndex((page) => page.pathname.includes('eholdings') && page.pathname !== location.pathname);
+
+    if (nonEholdingsPagesCount > 0) {
+      // go(-1) will return to previous page
+      // if previous page was non eholdings - need to call go(-2)
+      // and this extends to N non-eholdings pages - call go(-(N + 1))
+      history.go(-(nonEholdingsPagesCount + 1));
```

Comment:
If we do many navigations on the non eholdings page, then the navigation process will be unexpected on the eholdings page. 

I think we need to take into account the number of navigations on a non eholdings page too. It might be fixed by passing `history.length` via `LastVisitedContext` that is used in [main navigation](https://github.com/folio-org/stripes-core/blob/bbdf2dd352fbbaeeb91ab44c13351fa5b55ca5b4/src/components/MainNav/MainNav.js#L200).


https://user-images.githubusercontent.com/77053927/210651360-4a676aad-4905-4f2a-a597-16f557b7f95d.mp4